Form/ChoiceList/PhpcrOdmQueryBuilderLoader.php

```diff
@@ -93,8 +94,13 @@ public function getEntitiesByIds($identifier, array $values)
         $qb = clone $this->queryBuilder;
         $alias = $qb->getPrimaryAlias();
         $where = $qb->andWhere()->orX();
+
         foreach ($values as $val) {
-            $where->same($val, $alias);
+            if (UUIDHelper::isUUID($val)) {
+                $where->eq()->field($alias.'.uuid')->literal($val)->end();
```

Comment:
@dbu The class is always in the query builder, but i think we need to use reflection in order to get the class out of it. The issue is more with the tests in `PhpcrOdmQueryBuilderLoaderTest.php`, there the `$class` is never passed, so I could fix this to add a class to the 3rth argument of its constructor.